thundra/wrappers/fastapi/fastapi_wrapper.py

```diff
@@ -0,0 +1,108 @@
+from functools import wraps
+import logging
+from thundra import constants
+
+from thundra.opentracing.tracer import ThundraTracer
+from opentracing.scope_managers.asyncio import AsyncioScopeManager
+from thundra.wrappers.base_wrapper import BaseWrapper
+
+from thundra.config.config_provider import ConfigProvider
+from thundra.context.execution_context_manager import ExecutionContextManager
+from thundra.context.plugin_context import PluginContext
+from thundra.application.global_application_info_provider import GlobalApplicationInfoProvider
+from thundra.context.tracing_execution_context_provider import TracingExecutionContextProvider
+from thundra.wrappers.fastapi import fastapi_executor
+from thundra.config import config_names
+from thundra.wrappers import wrapper_utils, web_wrapper_utils
+import thundra.utils as utils
+
+import traceback
+
+
+logger = logging.getLogger(__name__)
+
+
+class FastapiWrapper(BaseWrapper):
+
+    def __init__(self, api_key=None, disable_trace=False, disable_metric=True, disable_log=True, opts=None):
+        super(FastapiWrapper, self).__init__(api_key, disable_trace, disable_metric, disable_log, opts)
+        self.application_info_provider = GlobalApplicationInfoProvider()
+        ThundraTracer.create_instance(AsyncioScopeManager())
+        ExecutionContextManager.set_provider(TracingExecutionContextProvider())
+        self.plugin_context = PluginContext(application_info=self.application_info_provider.get_application_info(),
+                                            request_count=0,
+                                            executor=fastapi_executor,
+                                            api_key=self.api_key)
+
+        self.plugins = wrapper_utils.initialize_plugins(self.plugin_context, disable_trace, disable_metric, disable_log,
+                                                        config=self.config)
+
+        web_wrapper_utils.update_application_info(self.application_info_provider, self.plugin_context.application_info,
+                                                  constants.ClassNames['FASTAPI'])
+
+
+    def before_request(self, request, req_body):
+        execution_context = wrapper_utils.create_execution_context()
+        execution_context.platform_data["request"] = request
+        execution_context.platform_data["request"]["body"] = req_body
+
+        self.plugin_context.request_count += 1
+        self.execute_hook("before:invocation", execution_context)
+
+        return execution_context
+
+
+    def after_request(self, execution_context):
+        self.prepare_and_send_reports_async(execution_context)
+
+    def error_handler(self, error):
```

Comment:
It might be safer to get the execution context as passed 2nd argument by caller below instead of getting by `ExecutionContextManager.get()`

Reply:
done.